### DISCUSSION TITLE
Database backup and restoration

### DISCUSSION THREAD

**Discussion by lyhnet** [immich-app/immich#4565] [Q&A] [answered]
Hi

I read (https://documentation.immich.app/docs/administration/backup-and-restore) where it is stated:

"Immich saves [file paths in the database](https://github.com/immich-app/immich/discussions/3299), it does not scan the library folder to update the database so backups are crucial."

I would like to know if I understand this correctly. 

1. If I take a backup of the database Now.
2. I add a photo via Upload
3. I confirm that the file is detected and processed into my library
4. Something happens to my Immich database (say I delete it by accident).
5. I make a fresh install of Immich (it should never have run before restore)
6. Restore the latest backup from step 1. 
7. The photo uploaded in step 2 will never be part of my library. Its physically in the library folder of my user. But since the database only holds file paths and does not scan that photo will never discovered by Immich

Is this correct?

If yes, is this an acceptable approach? 

One solution should be to use immich-cli upload (https://documentation.immich.app/docs/features/bulk-upload) to copy all files later than the latest database backup. But a scan function in Immich would be nice. 

-Anders

**Comment by alextran1502** [comment]
If you want to use the scan mechanism, you can use the new library feature here. https://immich.app/docs/features/libraries

Let me know if this serves your needs

**Reply by lyhnet** [reply]
I think jsram91 [comment-7344538](/https://github.com/immich-app/immich/discussions/4565#discussioncomment-7344538) in the last paragraph explained what I was missing.

**Comment by jrasm91** [comment, marked as answer]
Fwiw, the new repair screen would tell you about the untracked files and then you could re-upload them and you would be good to go.

When it comes to backups the issue you described is not unique to immich. You can investigate alternative methods to achieve continuous backups if your tolerance for data loss warrants that. At a minimum we recommend periodic database backups.

**Reply by lyhnet** [reply]
yes exactly. The scans, as I understand it, tests if the asset path exists. 

I would suggest to make a scan of the files in the Default library and create assets for asset-less files.
